files/build_templates/docker_image_ctl.j2

```diff
@@ -168,6 +168,9 @@ wait() {
 }
 
 stop() {
+{%- if docker_container_name == "pmon" %}
+    docker exec pmon post-syseeprom -c
```

Comment:
>docker exec pmon post-syseeprom -c [](start = 4, length = 34)

Could you clarify the motivation of 'clear' StateDB? Seems some logic inside container, why handle it outside?

Reply:
@qiluo-msft, the motivation to clear DB is that in the case config reload or reboot, the info written to state DB should be cleared.  I placed the logic here because as you may know "post-syseeprom -w" is a one-shot task, after writing syseeprom info to state DB it will exit. So has to call "post-syseeprom -c" somewhere to clear the DB, I didn't find a better place inside pmon to do this job, so placed it here. Do you have some suggestions for a better place to execute this command?

Reply:
Thanks! I believe the cleanup before docker stop is quite generic requirement. Let's assume all the processes inside container are supervisord controlled
1. If the process is long running, the cleanup should be inside the process as a signal handler
2. If the process is one-shot, the cleanup should be inside docker, and supervisord itself could know the signal sent by docker engine.

The I suggest you explorer http://supervisord.org/events.html#supervisor-state-change-stopping-event-type for the second case. No matter what is the final solution,  I am looking forward it reusable in other containers.

Reply:
@qiluo-msft thanks for the suggestion, will take a look and update.

Reply:
@qiluo-msft  @jleveque  I prototyped a small event listener to delete the DB on receiving Event "SUPERVISOR_STATE_CHANGE_STOPPING", it works.  So seems to me the solution will be adding a small event listener daemon to pmon container. please share your view.

Reply:
Thanks for the exploration! Please go ahead.

Reply:
I think the event listener could be a good solution.

Reply:
@qiluo-msft @jleveque please check the code for event listener in PR https://github.com/Azure/sonic-platform-daemons/pull/28.  And also this PR was updated with make file and supervisord conf for event listener.